backend/model/top_container.rb

```diff
@@ -265,6 +265,50 @@ def self.batch_update(ids, fields)
   end
 
 
+  def self.bulk_update_container_profile(ids, container_profile_uri)
```

Comment:
Rejigged this method in a separate commit to get rid of the relationship implementation details.  The new version won't be quite as fast (boo) but I think sticking to using the APIs is worth the trade-off.